prac_09/sort_files_v2.py

```diff
@@ -0,0 +1,32 @@
+"""
+CP1404/CP5632 Practical
+Program to sort files into directories based user input
+By Christopher Caferra
+"""
+
+import  os
+import shutil
+
+
+def main():
+    """Program to sort file extensions based on categories chosen by the user"""
+    extensions = {}
+    os.chdir('FilesToSort')
+    for filename in os.listdir('.'):        # save file names to list and extensions to dictionary
+        file_type = filename[filename.find('.') + 1:]
+        if filename.find(file_type):
+            extensions[file_type] = ""
+    for extension in extensions:     # get file category for every extension
+        category = input("What category would you like to sort {} files into? ".format(extension))
+        extensions[extension] = category
+    for category in extensions.values():    # make file category folder if it does not exist already
+        if not os.path.isdir(category):
+            os.mkdir(category)
+            print(category)
+    for filename in os.listdir('.'):                  # move files into category folders
+        file_type = filename[filename.find('.') + 1:]
+        if filename.find(file_type):
+            shutil.move(filename, extensions.get(file_type) + "/" + filename)
```

Comment:
Extensions[file_type] works the same as extensions.get(file_type) and it's more just personal preference but may be better overall (since it also shortens the code). Dict.get() is mainly useful if you are not sure that aa value is going to be returned and so you can set a default if you wish (whereas the other method returns an error (useful if you want to make sure that the None type isn't returned if it doesn't exist))